pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -178,7 +186,13 @@ func NewCompactor(cfg Config, objectClient client.ObjectClient, schemaConfig con
 		defaultRetentionStats.Set(limits.DefaultLimits().RetentionPeriod.String())
 	}
 	if cfg.SharedStoreType == "" {
-		return nil, errors.New("compactor shared_store_type must be specified")
+		objectTypes := make([]string, len(objectClients))
+		i := 0
+		for key := range objectClients {
+			objectTypes[i] = key
+			i++
+		}
+		level.Info(util_log.Logger).Log("msg", "compactor shared_store_type not specified. Initializing compactor to operator on the following object stores", "stores", objectTypes)
```

Comment:
Does this get printed correctly? I'd imagine we'd need to concat all the values of the slice...?

Reply:
oh, I overlooked it. Will check